pkg/storage/unified/resource/index.go

```diff
@@ -462,29 +465,42 @@ func createInMemoryIndex() (bleve.Index, string, error) {
 	return index, "", err
 }
 
+type IndexerListOptions struct {
+	*ListOptions
+	Kind string
+}
+
 // TODO - fetch from api
-func fetchResourceTypes() []*ListOptions {
-	items := []*ListOptions{}
-	items = append(items,
-		&ListOptions{
-			Key: &ResourceKey{
-				Group:    "playlist.grafana.app",
-				Resource: "playlists",
+func fetchResourceTypes() []*IndexerListOptions {
+	return []*IndexerListOptions{
+		{
+			ListOptions: &ListOptions{
+				Key: &ResourceKey{
+					Group:    "playlist.grafana.app",
+					Resource: "playlists",
+				},
 			},
+			Kind: "Playlist",
 		},
-		&ListOptions{
-			Key: &ResourceKey{
-				Group:    "folder.grafana.app",
-				Resource: "folders",
+		{
+			ListOptions: &ListOptions{
+				Key: &ResourceKey{
+					Group:    "dashboard.grafana.app",
+					Resource: "dashboards",
+				},
 			},
+			Kind: "Dashboard",
 		},
-		&ListOptions{
-			Key: &ResourceKey{
-				Group:    "dashboard.grafana.app",
-				Resource: "dashboards",
+		{
```

Comment:
we will have to index folder first.  when indexing dashboards we will need to lookup folders in the index

Reply:
ah okay, interesting. I'll change the order and leave a comment 👍